omf-python/src/file/reader.rs

```diff
@@ -0,0 +1,277 @@
+use crate::array::{
+    PyBooleanArray, PyBoundaryArray, PyColorArray, PyGradientArray, PyImageArray, PyIndexArray,
+    PyNameArray, PyNumberArray, PyScalarArray, PySegmentArray, PyTexcoordArray, PyTextArray,
+    PyTriangleArray, PyVectorArray, PyVertexArray,
+};
+use crate::errors::OmfException;
+use crate::validate::PyProblem;
+use crate::PyProject;
+use omf::error::Error::IoError;
+use omf::file::{Limits, Reader};
+use omf::Color;
+use pyo3::prelude::*;
+use pyo3::types::PyBytes;
+use pyo3_stub_gen::derive::*;
+use std::fs::File;
+
+#[gen_stub_pyclass]
+#[pyclass(name = "Limits")]
+/// Memory limits for reading OMF files.
+pub struct PyLimits {
+    #[pyo3(get, set)]
+    pub json_bytes: Option<u64>,
+    #[pyo3(get, set)]
+    pub image_bytes: Option<u64>,
+    #[pyo3(get, set)]
+    pub image_dim: Option<u32>,
+    #[pyo3(get, set)]
+    pub validation: Option<u32>,
+}
+
```

Comment:
It can be tidier to provide `From` implementations for the conversions back and forth from the wrapped types. This code is untested:
```suggestion
impl From<Limits> for PyLimits {
    fn from(limits: Limits) -> Self {
        Self {
            json_bytes: limits.json_bytes,
            image_bytes: limits.image_bytes,
            image_dim: limits.image_dim,
            validation: limits.validation,
        }
    }
}

impl From<PyLimits> for Limits {
    fn from(py_limits: PyLimits) -> Self {
        Self {
            json_bytes: py_limits.json_bytes,
            image_bytes: py_limits.image_bytes,
            image_dim: py_limits.image_dim,
            validation: py_limits.validation,
        }
    }
}
```
Then anywhere that needs to convert between the two can just use `limits.into()`.